### PR TITLE
slightly adjust the host_match api

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
tests fail because this depends on #40

File: lib/katello_attach_subscription/host_matcher.rb [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+      host['type'] = KatelloAttachSubscription::FactAnalyzer.system_type(host) unless host['type']
       valid = true
       valid = false unless self.match_hostname(config['hostname'], host['name'])
       valid = false unless self.match_type(config['type'], host['type'])
       host_facts = host['facts'] || {}
-      config.fetch('match', []).each do |match|
+      config.fetch('facts', []).each do |match|
         matcher = match['matcher'] || 'string'
-        valid = false unless self.match_matcher(match['value'], host_facts[match['fact']], matcher)
+        valid = false unless self.match_matcher(match['value'], host_facts[match['name']], matcher)
```
Comment (mossicrue) [inline lib/katello_attach_subscription/host_matcher.rb:11]: valid is never true, also when match_matcher return true
I suggest to change in `valid = self.match_matcher(match['value'], host_facts[match['name']], matcher)`

Comment (evgeni) [reply lib/katello_attach_subscription/host_matcher.rb:11]: `valid` starts out as `true` in line 5. afterwards it's flipped to `false` when one of the matchers fail. this is successfully tested by the `test_host_rhel_rhv_vm` test

**Comment by evgeni** [comment]
ACK'ed by @mossicrue on IRC

**Pull request merged**